src/app/core/dto/default-workspace.ts

```diff
@@ -0,0 +1,27 @@
+/*
+ * Lumeer: Modern Data Definition and Processing Platform
+ *
+ * Copyright (C) since 2017 Answer Institute, s.r.o. and/or its affiliates.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+export interface DefaultWorkspace {
+
+  organizationCode?: string;
+  organizationId: string;
+  projectCode?: string;
+  projectId: string;
```

Comment:
Why do we need both codes and IDs?

Reply:
Because we are sending `ids` but receive both. Codes are computed before send, so when organization and project is changed, we send fresh code for navigate to correct url.